pkg/cloud/services.go

```diff
@@ -275,21 +258,81 @@ func (c *AwsServiceClient) GetServiceList(filter *FilterServicesInput) []Service
 	return services
 }
 
-func (c *AwsServiceClient) GetListingFromArn(arn *string) ServiceListing {
-	r, err := regexp.Compile("([^\\/]*)\\/([^-]*)(-.*)?")
+func (c *AwsServiceClient) GetListingFromArn(arn *string) (*ServiceListing, error) {
+	svc, err := c.client.DescribeServices(&ecs.DescribeServicesInput{
+		Cluster: &c.clusterName,
+		Include: []*string{
+			mdl.String(ecs.ServiceFieldTags),
+		},
+		Services: []*string{
+			arn,
+		},
+	})
 
 	if err != nil {
-		panic(err)
+		return nil, err
+	}
+
+	if len(svc.Services) > 1 {
+		return nil, fmt.Errorf("there is more than one service with the specified arn")
+	}
+
+	if len(svc.Services) == 0 {
+		return nil, fmt.Errorf("there is no service with the specified arn")
 	}
 
-	matches := r.FindStringSubmatch(*arn)
+	return c.GetListingFromService(svc.Services[0]), nil
+}
+
+func (c *AwsServiceClient) GetListingFromService(svc *ecs.Service) *ServiceListing {
+	tags := map[string]string{}
+	funk.ForEach(svc.Tags, func(tag *ecs.Tag) {
+		tags[*tag.Key] = *tag.Value
+	})
 
-	listing := ServiceListing{
-		Arn:         matches[0],
-		Name:        matches[2] + matches[3],
-		Application: matches[2],
-		EventType:   strings.TrimPrefix(matches[3], "-"),
+	listing := &ServiceListing{
+		Arn:  *svc.ServiceArn,
+		Name: *svc.ServiceName,
+		Tags: tags,
 	}
 
 	return listing
 }
+
+func (c *AwsServiceClient) getLoggerFields(filter *FilterServicesInput) mon.Fields {
+	fields := mon.Fields{}
+
+	for key, value := range filter.Tags {
+		fields[key] = value
+	}
+
+	return fields
+}
+
+func (c *AwsServiceClient) hasTags(listing *ServiceListing, tags map[string][]string) bool {
+	if listing == nil {
+		return false
+	}
+
+	for key, value := range tags {
+		v, ok := listing.Tags[key]
+
+		if !ok {
+			return false
+		}
+
+		foundValue := false
+
+		for _, subvalue := range value {
+			if subvalue == v {
+				foundValue = true
+			}
+		}
+
+		if !foundValue {
```

Comment:
do we need this? can we not just return foundValue?

Reply:
no, because we are in a loop and want to check that for each tag name there is one matching value

Reply:
makes sense